src/mixins/Registry.ts

```diff
@@ -17,18 +15,20 @@ export function RegistryMixin<T extends Constructor<WidgetBase<RegistryMixinProp
 
 		@diffProperty('registry')
 		public diffPropertyRegistry(previousValue: WidgetRegistry, value: WidgetRegistry): PropertyChangeRecord {
+			let changed = false;
+			if (!previousValue) {
+				this.registries.add(value);
+				changed = true;
+			}
+			else if (previousValue !== value) {
+				this.registries.replace(previousValue, value);
+				changed = true;
+			}
```

Comment:
Is there a branch here for comparing the length of registries where the previous and new that are the same instance?